pkg/localmetrics/localmetrics.go

```diff
@@ -16,51 +16,167 @@ package localmetrics
 
 import (
 	"net/http"
+	"net/url"
+	"strings"
 	"time"
 
-	dms "github.com/openshift/deadmanssnitch-operator/pkg/controller/deadmanssnitch"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 var (
-	MetricDeadMansSnitchHeartbeat = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "metricDeadMansSnitchHeartbeat",
-		Help: "Metric for heartbeating the Dead Man's Snitch api",
-	}, []string{"name"})
+	Collector *MetricsCollector
+)
 
-	MetricsList = []prometheus.Collector{
-		MetricDeadMansSnitchHeartbeat,
-	}
+const (
+	operatorName      = "deadmanssnitch-operator"
+	snitchMethodLabel = "method"
 )
 
-// UpdateMetrics updates all the metrics ever 5 minutes
-func UpdateMetrics() {
+type MetricsCollector struct {
+	ReconcileDuration  prometheus.Histogram
+	apiCallDuration    *prometheus.HistogramVec
+	snitchCallErrors   prometheus.Counter
+	snitchCallDuration *prometheus.HistogramVec
+	collectors         []prometheus.Collector
```

Comment:
Why is this here? I don't see it being used (except by `AddCollector`, which we also don't need, see below).

Reply:
Addressed in #69.